internal/pkg/telemetry/ddtracer.go

```diff
@@ -0,0 +1,101 @@
+package telemetry
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/xerrors"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	attrDDError           = attribute.Key("error")
+	attrDDErrorType       = attribute.Key("error.type")
+	attrDDErrorDetails    = attribute.Key("error.details")
+	attrDDErrorStackTrace = attribute.Key("error.stack")
+	stackTraceLength      = 20
+)
+
+type wrappedDDTracer struct {
+	tracer trace.Tracer
+}
+
+type wrappedDDSpan struct {
+	trace.Span
+}
+
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+func (s *wrappedDDSpan) RecordError(err error, _ ...trace.EventOption) {
+	errMsg := err.Error()
+	errType := typeStr(err)
+	s.Span.SetAttributes(
+		semconv.ExceptionMessage(errMsg),
+		semconv.ExceptionType(errType),
+		attrDDError.String(errMsg),
+		attrDDErrorType.String(errType),
+	)
+
+	// Add details
+	switch err.(type) { //nolint: errorlint
+	case xerrors.Formatter, fmt.Formatter:
+		s.Span.SetAttributes(attrDDErrorDetails.String(fmt.Sprintf("%+v", err)))
+	}
+
+	// Add stack trace
+	if v, ok := err.(stackTracer); ok { //nolint: errorlint
+		pcs := v.StackTrace()
+		s.Span.SetAttributes(attrDDErrorStackTrace.String(formatStackTrace(pcs, len(pcs))))
+	} else {
+		s.Span.SetAttributes(attrDDErrorStackTrace.String(takeStacktrace(stackTraceLength, 1)))
+	}
+}
```

Comment:
V OTEL DD chybalo uplne zaznamenavanie chyb, v takej podobe aby to bolo spravne zobrazene v DD UI.
Nastudoval som si kod DD (not OTEL) tracer a doplnil co treba.

Reply:
Vid `RecordError` - `no-op`: https://github.com/DataDog/dd-trace-go/blob/596353a907da3ce3f5c2e554e2ac252dffbf8913/ddtrace/opentelemetry/span.go#L35